control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1414,7 +1414,12 @@ func (r *reconciler) reconcileOLM(ctx context.Context, hcp *hyperv1.HostedContro
 		// Management OLM Placement
 		if _, err := r.CreateOrUpdate(ctx, r.client, operatorHub, func() error {
 			if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.OperatorHub != nil {
+				// if the spec.Configuration contains something related with OperatorHub configuration
+				// then we need to sync the OperatorHub.Spec to the spec.Configuration.OperatorHub
```

Comment:
nit: could be better rephrased, something like: `if spec.Configuration.OperatorHub is set, we need to sync it to the guest cluster`. I find this comment tricky to follow at first glance.